cpp/open3d/t/geometry/BoundingVolume.cpp

```
@@ -289,6 +324,346 @@ AxisAlignedBoundingBox AxisAlignedBoundingBox::FromLegacy(
+                     core::TensorKey::Index(2)}) *
+            extent_.GetItem({core::TensorKey::Index(2)}) * 0.5;
+
+    core::Tensor points = core::Tensor::Zeros({8, 3}, GetDtype(), GetDevice());
+    points.SetItem(core::TensorKey::Index(0),
+                   center_ - x_axis - y_axis - z_axis);
+    points.SetItem(core::TensorKey::Index(1),
+                   center_ + x_axis - y_axis - z_axis);
+    points.SetItem(core::TensorKey::Index(2),
+                   center_ - x_axis + y_axis - z_axis);
+    points.SetItem(core::TensorKey::Index(3),
+                   center_ - x_axis - y_axis + z_axis);
+    points.SetItem(core::TensorKey::Index(4),
+                   center_ + x_axis + y_axis + z_axis);
+    points.SetItem(core::TensorKey::Index(5),
+                   center_ - x_axis + y_axis + z_axis);
+    points.SetItem(core::TensorKey::Index(6),
+                   center_ + x_axis - y_axis + z_axis);
+    points.SetItem(core::TensorKey::Index(7),
+                   center_ + x_axis + y_axis - z_axis);
+    return points;
+}
+
+OrientedBoundingBox &OrientedBoundingBox::Translate(
+        const core::Tensor &translation, bool relative) {
+    core::AssertTensorDevice(translation, GetDevice());
+    core::AssertTensorShape(translation, {3});
+    if (translation.GetDtype() != GetDtype()) {
+        PrintWarningMessageFromTensor("translation", translation.GetDtype(),
+                                      GetDtype());
+    }
+
+    const core::Tensor translation_d = translation.To(GetDtype());
+    if (relative) {
+        center_ += translation_d;
+    } else {
+        center_ = translation_d;
+    }
+    return *this;
+}
+
+OrientedBoundingBox &OrientedBoundingBox::Rotate(
+        const core::Tensor &rotation,
+        const utility::optional<core::Tensor> &center) {
+    core::AssertTensorDevice(rotation, GetDevice());
+    core::AssertTensorShape(rotation, {3, 3});
+    if (rotation.GetDtype() != GetDtype()) {
```

Skip warning, silently convert.